src/features/apiDelegation/offered/ChooseApiPage/ChooseApiPage.tsx

```diff
@@ -78,7 +84,10 @@ export const ChooseApiPage = () => {
 
     for (const key of urlParams.keys()) {
       searchResults?.forEach((api: DelegableApi) => {
-        if (api.identifier === key) {
+        if (
+          api.identifier === key &&
+          chosenApis.filter((chosenApi: DelegableApi) => chosenApi.identifier === key).length === 0
+        ) {
```

Comment:
Great performance improvent. I changed the dependency array to isLoading and it works. 
We still need to check if the api identifier is in the list to prevent doubble-rendering items, so i'll keep that in as well.